### PR TITLE
Added a custom description for each metric in the checkstyle plugin

### PR DESCRIPTION
**Pull request by maximAtanasov**
Closes issues #35 and #14.

### CODE REVIEW

File: coderadar-core/src/main/java/io/reflectoring/coderadar/query/service/GetFileContentWithMetricsService.java [hunk 1]
```diff
@@ -24,7 +25,8 @@ public GetFileContentWithMetricsService(
       GetProjectPort getProjectPort,
       GetRawCommitContentPort getRawCommitContentPort,
       GetMetricsAndFindingsForFilePort getMetricsAndFindingsForFilePort,
-      CoderadarConfigurationProperties coderadarConfigurationProperties) {
+      CoderadarConfigurationProperties coderadarConfigurationProperties,
+      GetMetricDescriptionService getMetricDescriptionService) {
```
[the remark this answers is not in the document]
Comment (maximAtanasov) [reply coderadar-core/src/main/java/io/reflectoring/coderadar/query/service/GetFileContentWithMetricsService.java:29]: Done.

File: coderadar-plugins/checkstyle-analyzer-plugin/src/main/java/io/reflectoring/coderadar/analyzer/checkstyle/CheckstyleSourceCodeFileAnalyzerPlugin.java [hunk 2]
```diff
@@ -17,6 +17,8 @@
     implements SourceCodeFileAnalyzerPlugin, ConfigurableAnalyzerPlugin {
 
   private Logger logger = LoggerFactory.getLogger(CheckstyleSourceCodeFileAnalyzerPlugin.class);
```
[the remark this answers is not in the document]
Comment (maximAtanasov) [reply coderadar-plugins/checkstyle-analyzer-plugin/src/main/java/io/reflectoring/coderadar/analyzer/checkstyle/CheckstyleSourceCodeFileAnalyzerPlugin.java:19]: Done.

File: coderadar-plugins/checkstyle-analyzer-plugin/src/main/java/io/reflectoring/coderadar/analyzer/checkstyle/CheckstyleMetricDescriptions.java [hunk 3]
```diff
@@ -0,0 +1,452 @@
...
+
+  public static final String WHITESPACE_AFTER = "Add a whitespace after this comma.";
+
+  public static final String WHITESPACE_AROUND = "Missing whitespace.";
+
+  public static final String WRITE_TAG = "WriteTag."; // Let's ignore this one for now
+
+  public static final String OUTER_TYPE_FILENAME = "Class name does not match file name.";
+
+  public static final String CUSTOM_IMPORT_ORDER = "Incorrect import order.";
```
[the remark this answers is not in the document]
Comment (maximAtanasov) [reply coderadar-plugins/checkstyle-analyzer-plugin/src/main/java/io/reflectoring/coderadar/analyzer/checkstyle/CheckstyleMetricDescriptions.java:307]: Done.

**Comment by maximAtanasov** [comment]
@KilianKrause Checkstyle descriptions are now displayed instead of the custom ones.

File: coderadar-core/src/main/java/io/reflectoring/coderadar/contributor/port/driver/GetContributorsForPathCommand.java [hunk 4]
```diff
@@ -9,6 +9,6 @@
 @NoArgsConstructor
 @AllArgsConstructor
 public class GetContributorsForPathCommand {
-  @NotBlank private String path;
+  private String path;
```
Comment (KilianKrause) [inline coderadar-core/src/main/java/io/reflectoring/coderadar/contributor/port/driver/GetContributorsForPathCommand.java:12]: Why did you remove the `@NotBlank` annotation?

File: coderadar-plugin-api/src/main/java/io/reflectoring/coderadar/plugin/api/Finding.java [hunk 5]
```diff
@@ -16,13 +16,16 @@
...
+
     /**
      * Marks a finding over multiple complete lines.
      *
      * @param lineStart the line number where the finding starts.
      * @param lineEnd   the line number where the finding ends.
+     * @param message the message of the metric.
      */
-    public Finding(Integer lineStart, Integer lineEnd) {
+    public Finding(Integer lineStart, Integer lineEnd, String message) {
```
Comment (KilianKrause) [inline coderadar-plugin-api/src/main/java/io/reflectoring/coderadar/plugin/api/Finding.java:28]: The parameter `message` is not used here. Also, the constructor is never used.

File: coderadar-core/src/main/java/io/reflectoring/coderadar/contributor/port/driver/GetContributorsForPathCommand.java (hunk 4 above)
Comment (maximAtanasov) [reply coderadar-core/src/main/java/io/reflectoring/coderadar/contributor/port/driver/GetContributorsForPathCommand.java]: It was a way to fix bug, but I've fixed it properly now.

File: coderadar-plugin-api/src/main/java/io/reflectoring/coderadar/plugin/api/Finding.java (hunk 5 above)
Comment (maximAtanasov) [reply coderadar-plugin-api/src/main/java/io/reflectoring/coderadar/plugin/api/Finding.java]: Done
